x/spec/types/spec.go

```diff
@@ -7,6 +7,97 @@ import (
 
 const minCU = 1
 
+type getterFunc func(index string) (Spec, bool)
+ 
+// ExpandSpec expands the "imports" field of the spec -if needed, recursively-
+// and adds to the current Spec those additional APIs from the imported Spec(s).
+// It takes as argument a getter function to fetch imported Spec(s) by name from
+// the keeper store, on demand. (Can't import the keeper module due to circular
+// dependencies)
+func (spec *Spec) ExpandSpec(getSpec getterFunc) (map[string]string, error) {
+	depends := map[string]bool{spec.Index: true}
+	details := map[string]string{"imports": fmt.Sprintf("%s", spec.Index)}
```

Comment:
why do we need this as a map? it looks like it contains only one key value pair

Reply:
That's what the caller expects, convention for reporting details of errors/events for printing.
Internally in `doExpandSpec()` we can maintain a single string - will fix.